tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/GitHubWebhookProcessor.cs

```diff
@@ -49,13 +49,26 @@ public async Task ProcessWebhookAsync(string eventName, string json, ILogger log
         {
             await Policy
                 .Handle<AbuseException>()
+                .Or<RateLimitExceededException>()
                 .RetryAsync(3, async (ex, retryCount) =>
                 {
-                    logger.LogWarning("Abuse exception detected, attempting retry.");
-                    var abuseEx = (AbuseException)ex;
+                    TimeSpan retryDelay = TimeSpan.FromSeconds(10); // Default.
+
+                    switch (ex)
+                    {
+                        case AbuseException abuseException:
+                            logger.LogWarning("Abuse exception detected, attempting retry.");
+                            retryDelay = TimeSpan.FromSeconds((double)abuseException.RetryAfterSeconds);
```

Comment:
Does this always give a valid value? 

BTW I assume you have read the guidelines at https://developer.github.com/v3/guides/best-practices-for-integrators/#dealing-with-abuse-rate-limits. If we are using our application id and not doing concurrent requests we shouldn't hit this abuse limit.

Reply:
I'd read those previously, but was probably time for a refresher. Thanks.

I think that the RetryAfterSeconds thing should always be populated in our case because we are only adding check runs (don't think that is classified as making content). In my experience the number of AbuseExceptions that made it threw was relatively small. Its the RateLimitExeceededExceptions which I wasn't handling which were more prevalent in our logs.

I think doing this should get us back into a healthy state even if there are times that our concurrent request counts break the rules. If it continues to be the problem I can get more aggressive with the rate limiting.